File: Daily Challanges/2026-06-June/2026-06-03.py
def is_valid_schema(obj):
    return (
        isinstance(obj, dict)
        and isinstance(obj.get("username"), str)
        and isinstance(obj.get("posts"), (int, float))
        and isinstance(obj.get("verified"), bool)
        and obj.get("role") in {
            "user",
            "creator",
            "moderator",
            "staff",
            "admin"
        }
    )

print(is_valid_schema({"username": "henry", "posts": 0, "verified": True, "role": "staff"})) 
print(is_valid_schema({"username": "sara", "posts": 45, "verified": False, "role": "creator", "followers": 70})) 

File: Daily Challanges/2026-06-June/2026-06-03.js
function isValidSchema(obj) {
  return (
    typeof obj === "object" &&
    obj !== null &&
    typeof obj.username === "string" &&
    typeof obj.posts === "number" &&
    typeof obj.verified === "boolean" &&
    ["user", "creator", "moderator", "staff", "admin"].includes(obj.role)
  );
}

console.log(
  isValidSchema({
    username: "henry",
    posts: 0,
    verified: true,
    role: "staff",
  }),
);

console.log(
  isValidSchema({
    username: "sara",
    posts: 45,
    verified: false,
    role: "creator",
    followers: 70,
  }),
);
